### PR TITLE
Add support for macOS with Apple Silicon

### PR DESCRIPTION
**Pull request by externl**

### CODE REVIEW

**Review by pepone — APPROVED** [review]
looks good

File: config/Make.rules.Darwin [hunk 1]
```diff
@@ -4,8 +4,10 @@
 
 supported-platforms             = macosx iphoneos iphonesimulator
 
+macos_arch                      ?= $(shell uname -m)
+
 macosx_ar                       = libtool
-macosx_cppflags                 = -mmacosx-version-min=10.14 -arch x86_64
+macosx_cppflags                 = -mmacosx-version-min=10.14 -arch $(macos_arch)
```
Comment (bentoi) [inline config/Make.rules.Darwin:10]: Shouldn't we build a library with both Intel and ARM arch?

Comment (externl) [reply config/Make.rules.Darwin:10]: We could, I don't have strong feelings either way.

Comment (externl) [reply config/Make.rules.Darwin:10]: I'm going to remove it, I think we should just use the default. Should we do the same for the iphone target

**Review by bentoi — APPROVED** [review]
Looks good.

**Pull request merged**